### PR TITLE
Enable tests with DOCS_GITHUB_TOKEN

### PR DESCRIPTION
**Pull request by yufeih**
Removed a couple of corner case restore test to reduce the amount of secret management.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/docfx/pull/7091)

### CODE REVIEW

File: docs/specs/restore.yml [hunk 1]
```diff
@@ -365,116 +365,3 @@ outputs:
         "docs/a.md": [{ "source": "docs/b.md", "type": "file" }]
       }
     }
----
-# Show error for system service account SSO issue
```
Comment (mingwli) [inline docs/specs/restore.yml:369]: can we simply `skip`, incase adding them back?

**Review by mingwli — APPROVED** [review]
